### ISSUE TITLE
Can't trigger factoids containing : without whitespace preceding it

### ISSUE THREAD

**Issue by dgw** [zigdon/xkcd-Bucket#76]
Clearly something to do with how Bucket strips off the first bit for use as `$to` in the tidbit. With #75 enabled on my instance, pinning this down was pretty easy.

What's less clear is the way ahead. For my particular case it would work simply to avoid stripping off the first part of the line if there's no space following the colon…but what of legitimate triggers of the form `Word: More Stuff`?

Basically, is it worth e.g. checking to see if the potential value of `$to` is in the channel before stripping it from the line start? Or have I crossed into truly esoteric territory that should be left alone?

**Comment by zigdon** [comment]
Don't we already keep a list of channel membership, for use in $someone? In that case, checking to see if the addressee is in that list shouldn't be too bad.

**Comment by dgw** [comment]
I put together an implementation of that solution, and a couple quick tests show it's working. I'll let it stew in production for a while before posting a PR, per usual, to shake out as many bugs as I can.

**Comment by dgw** [comment]
Probably good that I didn't PR/merge this yet… I'm having trouble triggering factoids with the `$to` mechanism while running this patch, I think. More work needed…

**Comment by dgw** [comment]
Testing a new, simpler implementation. It's all committed and ready to PR once I put it through another week or two of testing—probably two, since it took me that long to find the bug in my original naïve implementation.
